Catch NotImplementedError when trying to sleep with geth

christianb93 opened the pull request:
### What I did

Related issue: 1227

### How I did it

In network/state.py, add a simple try / except block around the sleep statement in `_add_to_undo_buffer`, exactly as already present in `_revert`.

### How to verify it

Run geth v1.10.8
Then, in a separate terminal, run `brownie console` and test a simple transfer:

```
dev = accounts[0]
alice = accounts.add()
dev.transfer(to=alice, amount="1 ether")
```
This should not raise any errors, and `alice.balance()` should show that the Ether has been transferred. Unit tests in tests/network/state still pass.

### Checklist

- [x ] I have confirmed that my PR passes all linting checks
- [ ] I have included test cases
- [ ] I have updated the documentation
- [ ] I have added an entry to the changelog

christianb93 commented:
Update: also added a unit test case for this

iamdefinitelyahuman commented on tests/network/account/test_account_transfer.py:
```diff
@@ -31,6 +32,26 @@ def test_to_account(accounts):
+    assert(rpc.is_active())
+    monkeypatch.setattr(rpc, "sleep", _raises)
+    tx = accounts[0].transfer(accounts[1], 10000)
+    assert str(tx.receiver == accounts[1].address)
```
```suggestion
    assert str(tx.receiver) == accounts[1].address
```

iamdefinitelyahuman reviewed (commented):
Looks good, one incorrect assertion in the test tho and needs to be linted.

christianb93 commented:
Uh, stupid typo - thanks for catching this. I fixed this and corrected the linting error, PR is updated and latest changes from master have been merged into it (my first contribution, I hope I got this right)
